omnigibson/utils/motion_planning_utils.py

```diff
@@ -544,3 +545,76 @@ def detect_robot_collision_in_sim(robot, filter_objs=[], ignore_obj_in_hand=True
         if col_obj.category in filter_categories:
             collision_prims.remove(col_prim)
     return len(collision_prims) > 0
+
+def astar(search_map, start, goal, eight_connected=True):
+    """
+    A* search algorithm for finding a path from start to goal on a grid map
+
+    Args:
+        search_map (Array): 2D Grid map to search on
+        start (Array): Start position on the map
+        goal (Array): Goal position on the map
```

Comment:
what is `eight_connected`?

Reply:
whether we consider just the sides as neighbors or the diagonals too. let's document here.